Add Structural Search and Replace doc (#3275)
rust-analyzer/rust-analyzer: docs/user/features.md

Stop `cargo watch`.

#### Structural Seach and Replace 

Search and replace with named wildcards that will match any expression. 

> The syntax for a structural search replace command is `<search_pattern> ==>> <replace_pattern>` where `$<name>:expr` in the search pattern will match any expression and `$<name>` will fill the matching expression in the replace pattern. Available via the command `rust-analyer.ssr`.

Reviewer: ```suggestion
The syntax for a structural search replace command is `<search_pattern> ==>> <replace_pattern>`. You can use `$<name>:expr` in the search pattern to match any expression and `$<name>` will reference it in the replacement. Available via the command `rust-analyzer.ssr`.
```
